### ISSUE TITLE
add angular2 support

### ISSUE THREAD

**Issue by venil7** [satya164/quik#8]

**Comment by venil7** [comment]
automatic compilation of TypeScript files?

**Comment by satya164** [comment]
@venil7 Doesn't Angular have it's own CLI?

**Comment by venil7** [comment]
not AFAIK, it just seems like quik is very `react` centric, while it would be nice if it was library agnostic and be used with any es6 project. or maybe  a plugin system, so that any transpiler, like TypeScript can be plugged in..

**Comment by satya164** [comment]
@venil7 I surely want to add other things. Will need to make the React bit rock solid, and have a plugin system to extend the webpack config. Wanna help?

**Comment by satya164** [comment]
BTW there are two ways to extend `quik`,

1. Add custom loaders to the webpack config (there's no way to do this right now). This can be for simple things, like typescript support.
2. Write middlewares to handle different file types. See the [API docs](https://github.com/satya164/quik/blob/master/API.md). This can be for more sophisticated things, for example adding browserSync .

**Comment by venil7** [comment]
yes, but i thought that the main idea was to have it zero-setup
so maybe if option 1 was included out of the box

**Comment by satya164** [comment]
@venil7 Yeah. But the reason I want to add a plugin system is so support for new things can be added easily. Fo example, right now, to add typescript support, you've to change files in 3 places, the webpack config, the middleware which handles JS and the middleware which handles hot reload. It'll be better to have those changes self-contained in one file.

Also, people will not need to fork it if they want a slightly different feature set. They can just add something on top of it and publish as a separate module.

BTW I spent some time yesterday night trying to add typescript. The problem I faced was that the ts-loder was trying to require modules from current directory and not quik's node_modules directory. Once I sort it out, I'll add typescript. Help is welcome.

**Comment by venil7** [comment]
yeah, i agree on the plugin system, 
as long as they are installable via npm
